pkg/zfs/zfs_util.go

```diff
@@ -353,6 +378,51 @@ func DestroyVolume(vol *apis.ZFSVolume) error {
 	return nil
 }
 
+// CreateSnapshot creates the zfs volume snapshot
+func CreateSnapshot(snap *apis.ZFSSnapshot) error {
+
+	volume := snap.Labels[ZFSVolKey]
+	snapDataset := snap.Spec.PoolName + "/" + volume + "@" + snap.Name
+
+	if err := getVolume(snapDataset); err == nil {
+		logrus.Infof("snapshot already there %s", snapDataset)
+		// snapshot already there just return
+		return nil
+	}
+
+	args := buildZFSSnapCreateArgs(snap)
+	cmd := exec.Command(ZFSVolCmd, args...)
+	out, err := cmd.CombinedOutput()
+
+	if err != nil {
+		logrus.Errorf(
+			"zfs: could not create snapshot %v@%v cmd %v error: %s", volume, snap.Name, args, string(out),
+		)
+		return err
+	}
```

Comment:
Why are we logging if we are returning the error?

Reply:
for debugging purpose, to know the exact command fired and why that failed.

Reply:
So why are we not forming the entire error and returning? 
But I am fine.

Reply:
we can not have cmd args in the error, so logging here.